### PR TITLE
Update websocket.rst,remove WebSocketClient

### PR DESCRIPTION
**Pull request by Re-volution**
WebSocketServer and WebSocketServer no longer exist in godot4

### CODE REVIEW

File: tutorials/networking/websocket.rst [hunk 1]
```diff
@@ -38,59 +38,32 @@ This example will show you how to create a WebSocket connection to a remote serv
...
-        _client.connection_error.connect(_closed)
-        _client.connection_established.connect(_connected)
-        # This signal is emitted when not using the Multiplayer API every time
-        # a full packet is received.
-        # Alternatively, you could check get_peer(1).get_available_packets() in a loop.
-        _client.data_received.connect(_on_data)
-
         # Initiate connection to the given URL.
-        var err = _client.connect_to_url(websocket_url, ["lws-mirror-protocol"])
+        # remote ip addr and port
```
Comment (AThousandShips) [inline tutorials/networking/websocket.rst:46]:
```suggestion
        # Remote address and port.
```
Per the [comment style](https://docs.godotengine.org/en/latest/contributing/development/code_style_guidelines.html#comment-style-guide). 

Also there's no IP address provided in this example so specifying IP would be confusing

File: tutorials/networking/websocket.rst [hunk 2]
```diff
@@ -38,59 +38,32 @@ This example will show you how to create a WebSocket connection to a remote serv
...
+        var state = socket.get_ready_state()
+    	if state == WebSocketPeer.STATE_OPEN:
+    		while socket.get_available_packet_count():
+    			print("data packet：", socket.get_packet())
+    	elif state == WebSocketPeer.STATE_CLOSING:
+    		# closing
+    		pass
+    	elif state == WebSocketPeer.STATE_CLOSED:
+    		var code = socket.get_close_code()
+    		var reason = socket.get_close_reason()
```
Comment (AThousandShips) [inline tutorials/networking/websocket.rst:63]:
```suggestion
        var state = _client.get_ready_state()
    	if state == WebSocketPeer.STATE_OPEN:
    		while _client.get_available_packet_count():
    			print("data packet：", _client.get_packet())
    	elif state == WebSocketPeer.STATE_CLOSING:
    		# closing
    		pass
    	elif state == WebSocketPeer.STATE_CLOSED:
    		var code = _client.get_close_code()
    		var reason = _client.get_close_reason()
```
`socket` is not defined as a variable here, you probably forgot to update when copying from the documentation of `WebSocketPeer`

File: tutorials/networking/websocket.rst [hunk 3]
```diff
@@ -38,59 +38,32 @@ This example will show you how to create a WebSocket connection to a remote serv
...
     func _process(delta):
-        # Call this in _process or _physics_process. Data transfer, and signals
-        # emission will only happen when calling this function.
         _client.poll()
+        var state = socket.get_ready_state()
+    	if state == WebSocketPeer.STATE_OPEN:
+    		while socket.get_available_packet_count():
+    			print("data packet：", socket.get_packet())
+    	elif state == WebSocketPeer.STATE_CLOSING:
+    		# closing
```
Comment (AThousandShips) [inline tutorials/networking/websocket.rst:59]:
```suggestion
    		# Closing.
```

File: tutorials/networking/websocket.rst [hunk 4]
```diff
@@ -38,59 +38,32 @@ This example will show you how to create a WebSocket connection to a remote serv
 
     extends Node
 
-    # The URL we will connect to
-    export var websocket_url = "wss://libwebsockets.org"
```
Comment (AThousandShips) [inline tutorials/networking/websocket.rst:42]: Why was this removed?

**Review by AThousandShips — COMMENTED** [review]
The first part is now mostly redundant as it is almost exactly a copy of the code in the documentation for `WebSocketPeer`

File: tutorials/networking/websocket.rst [hunk 5]
```diff
@@ -38,59 +38,32 @@ This example will show you how to create a WebSocket connection to a remote serv
...
+    		while socket.get_available_packet_count():
+    			print("data packet：", socket.get_packet())
+    	elif state == WebSocketPeer.STATE_CLOSING:
+    		# closing
+    		pass
+    	elif state == WebSocketPeer.STATE_CLOSED:
+    		var code = socket.get_close_code()
+    		var reason = socket.get_close_reason()
+    		print("WebSocket close,code：%d,reason %s." % [code, reason])
+    		set_process(false) #  stop _process
```
Comment (AThousandShips) [inline tutorials/networking/websocket.rst:65]:
```suggestion
    		set_process(false) # Stop _process.
```

File: tutorials/networking/websocket.rst [hunk 6]
```diff
@@ -38,59 +38,32 @@ This example will show you how to create a WebSocket connection to a remote serv
...
-        print("Got data from server: ", _client.get_peer(1).get_packet().get_string_from_utf8())
-
     func _process(delta):
-        # Call this in _process or _physics_process. Data transfer, and signals
-        # emission will only happen when calling this function.
         _client.poll()
+        var state = socket.get_ready_state()
+    	if state == WebSocketPeer.STATE_OPEN:
+    		while socket.get_available_packet_count():
+    			print("data packet：", socket.get_packet())
```
Comment (AThousandShips) [inline tutorials/networking/websocket.rst:57]:
```suggestion
    			print("Data packet：", socket.get_packet())
```

File: tutorials/networking/websocket.rst [hunk 7]
```diff
@@ -38,59 +38,32 @@ This example will show you how to create a WebSocket connection to a remote serv
...
+    	elif state == WebSocketPeer.STATE_CLOSING:
+    		# closing
+    		pass
+    	elif state == WebSocketPeer.STATE_CLOSED:
+    		var code = socket.get_close_code()
+    		var reason = socket.get_close_reason()
+    		print("WebSocket close,code：%d,reason %s." % [code, reason])
```
Comment (AThousandShips) [inline tutorials/networking/websocket.rst:64]:
```suggestion
    		print("WebSocket closed with code: %d, reason %s." % [code, reason])
```

File: tutorials/networking/websocket.rst [hunk 8]
```diff
@@ -102,61 +75,54 @@ This example will show you how to create a WebSocket server that listens for rem
...
+
+    #handle msg
+    func _handlemsg():
+    	for key in dict :
+    		var peer = dict[key]
+    		peer.poll()
+    		var state = peer.get_ready_state()
+    		if state == WebSocketPeer.STATE_OPEN:#If the connection is closed too quickly,data cannot be read
+    			while peer.get_available_packet_count():
+    				print("packet：", peer.get_packet())
```
Comment (AThousandShips) [inline tutorials/networking/websocket.rst:113]:
```suggestion
    				print("Packet：", peer.get_packet())
```

File: tutorials/networking/websocket.rst [hunk 9]
```diff
@@ -102,61 +75,54 @@ This example will show you how to create a WebSocket server that listens for rem
...
-
+        print("Client %d disconnected % [id])
+
+    #handle msg
+    func _handlemsg():
+    	for key in dict :
+    		var peer = dict[key]
+    		peer.poll()
+    		var state = peer.get_ready_state()
+    		if state == WebSocketPeer.STATE_OPEN:#If the connection is closed too quickly,data cannot be read
```
Comment (AThousandShips) [inline tutorials/networking/websocket.rst:111]:
```suggestion
    		if state == WebSocketPeer.STATE_OPEN: # If the connection is closed too quickly, data cannot be read.
```

File: tutorials/networking/websocket.rst [hunk 10]
```diff
@@ -102,61 +75,54 @@ This example will show you how to create a WebSocket server that listens for rem
...
+    func _connected(id):
         # This is called when a new peer connects, "id" will be the assigned peer id,
-        # "proto" will be the selected WebSocket sub-protocol (which is optional)
-        print("Client %d connected with protocol: %s" % [id, proto])
-
-    func _close_request(id, code, reason):
-        # This is called when a client notifies that it wishes to close the connection,
-        # providing a reason string and close code.
-        print("Client %d disconnecting with code: %d, reason: %s" % [id, code, reason])
+        print("Client %d connected % [id])
```
Comment (AThousandShips) [inline tutorials/networking/websocket.rst:97]:
```suggestion
        print("Client %d connected." % [id])
```

**Comment by Piralein** [comment]
Please take [#7645](https://github.com/godotengine/godot-docs/pull/7645) into account.
We would like to prevent duplicate work.

**Comment by AThousandShips** [comment]
That's true had completely forgotten that one existed as well

File: tutorials/networking/websocket.rst (hunk 4 above)
Comment (Re-volution) [reply tutorials/networking/websocket.rst:42]: Because I tried and couldn't connect to this address

Comment (AThousandShips) [reply tutorials/networking/websocket.rst:42]: The problem is that this doesn't provide an example that can be tried to ensure you've done the right thing, your address below isn't a real address

Comment (Re-volution) [reply tutorials/networking/websocket.rst:42]: You're right, maybe this should be changed to `var websocket_url = "wss://localhost:8999"`, which would look better. It's best to add a comment # You need to start websocket server that listens to port 8999.

Comment (AThousandShips) [reply tutorials/networking/websocket.rst:42]: I think a proper working server is required, as is provided for the other PR changing this

Comment (AThousandShips) [reply tutorials/networking/websocket.rst:42]: #7645 is IMHO more clean, and predates this one, so I think this should be closed in favor of it

**Comment by AThousandShips** [comment]
Thank you for your contribution, sorry it didn't work out, hope to see other contributions from you in the future friend!
